### PR TITLE
fix logic error when unresgistering in ContractRegistry

### PR DESCRIPTION
**Pull request by hammeWang**

### CODE REVIEW

File: solidity/contracts/utility/ContractRegistry.sol [hunk 1]
```diff
@@ -98,8 +98,16 @@ contract ContractRegistry is IContractRegistry, Owned, Utils {
             items[_contractName].isSet = false;
 
             // if there are multiple items in the registry, move the last element to the deleted element's position
+            // and modify last element's registryItem.nameIndex in items,
+            // to point to the right position in contractNames
             if (contractNames.length > 1)
-                contractNames[items[_contractName].nameIndex] = contractNames[contractNames.length - 1];
+                string lastContractNameString = contractNames[contractNames.length - 1];
+                uint unresgisteredIndex = items[_contractName].nameIndex;
```
Comment (yudilevi) [inline solidity/contracts/utility/ContractRegistry.sol]: typo

Comment (hammeWang) [reply solidity/contracts/utility/ContractRegistry.sol]: could you be more specific?
I don't see anything wrong.

Comment (hammeWang) [reply solidity/contracts/utility/ContractRegistry.sol]: do you mean `unregisteredIndex` not ok?

Comment (yudilevi) [reply solidity/contracts/utility/ContractRegistry.sol]: yes, sorry `unresgisteredIndex` should be `unregisteredIndex`

Comment (hammeWang) [reply solidity/contracts/utility/ContractRegistry.sol]: @yudilevi thx, I have fixed it and commit again.

**Comment by yudilevi** [comment]
Also, please open a new pull request to the 0.4.4 branch as that's the next version branch.

Thanks!

**Comment by hammeWang** [comment]
@yudilevi OK. please check.

**Comment by yudilevi** [comment]
I see a new pull request from 0.4.4 into master, not into 0.4.4 - can you please look into that?
I can also merge your changes myself if you wish.

**Comment by hammeWang** [comment]
@yudilevi Could you merge for me? thx a lot.

**Comment by yudilevi** [comment]
Done - thank you :)

**Pull request closed without merging**